web/src/settings_profile_fields.ts

```diff
@@ -398,6 +399,76 @@ function disable_submit_btn_if_no_property_changed(
     );
 }
 
+function get_alphabetized_profile_field_choices(sortable_instance: SortableJS): string[] {
+    const choices_array: [string, string][] = [];
+    const empty_choices_array: [number, string][] = [];
+
+    const choices_id_array = sortable_instance.toArray();
+    for (const [array_index, choice_id] of choices_id_array.entries()) {
+        const choice_value = $(sortable_instance.el)
+            .find<HTMLInputElement>(`div[data-value="${choice_id}"] input`)
+            .val()!;
+
+        // Preserve position of empty fields by removing them from the array that we will sort.
+        // We insert these empty fields back at their original position later.
+        if (choice_value.length === 0) {
+            empty_choices_array.push([array_index, choice_id]);
+            continue;
+        }
+
+        choices_array.push([choice_value, choice_id]);
+    }
+
+    choices_array.sort((a, b) => a[0].localeCompare(b[0]));
+
+    for (const [choice_index, choice_id] of empty_choices_array) {
+        choices_array.splice(choice_index, 0, ["", choice_id]);
+    }
```

Comment:
Deciding positions for empty fields after sorting.
I think this deserves a discussion with Alya and Tim, so flagging it here. Although this approach of keeping them at original position looks good to me otherwise all these empty fields would align at the beginning.

Reply:
Following #design conversation, we now move empty fields to bottom when sorting.